### PR TITLE
Tests: Add utility class TuningConfigBuilder to make IndexTask tests more readable and concise

### PR DESCRIPTION
**Pull request by kfaraz**
### Changes

- No functional change
- Add class `TuningConfigBuilder` to build `IndexTuningConfig`, `CompactionTuningConfig`
- Remove old class `ParallelIndexTestingFactory.TuningConfigBuilder`
- Remove some unused fields and methods

### CODE REVIEW

File: extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/indexing/MSQCompactionRunnerTest.java [hunk 1]
```diff
@@ -397,39 +398,17 @@ private static CompactionTask.CompactionTuningConfig createTuningConfig(
...
-        null,
-        null,
-        null,
-        null
-    );
+    return TuningConfigBuilder
+        .forCompactionTask()
+        .withMaxRowsInMemory(500000)
+        .withMaxBytesInMemory(1000000L)
+        .withMaxTotalRows(Long.MAX_VALUE)
```
Comment (AmatyaAvadhanula) [inline extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/indexing/MSQCompactionRunnerTest.java:405]: Why is this needed?

**Review by AmatyaAvadhanula — APPROVED** [review]
Thanks for the changes, @kfaraz!

File: extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/indexing/MSQCompactionRunnerTest.java (hunk 1 above)
Comment (kfaraz) [reply extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/indexing/MSQCompactionRunnerTest.java:405]: Oh, might have been copied over from somewhere else.
Will get rid of it in a follow up if that's okay (to avoid re-triggering CI.)

Comment (AmatyaAvadhanula) [reply extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/indexing/MSQCompactionRunnerTest.java:405]: Sure!

**Comment by kfaraz** [comment]
Thanks a lot for the quick review, @AmatyaAvadhanula !
